pkg/controller/endpointslice/reconciler_test.go

```diff
@@ -460,34 +459,152 @@ func TestReconcile1Pod(t *testing.T) {
+			existingSlices := []*discovery.EndpointSlice{}
+			if tc.existing != nil {
+				existingSlices = append(existingSlices, tc.existing)
+				_, createErr := client.DiscoveryV1().EndpointSlices(namespace).Create(context.TODO(), tc.existing, metav1.CreateOptions{})
+				assert.Nil(t, createErr, "Expected no error creating endpoint slice")
+			}
 
-	assert.Regexp(t, "^"+svc.Name, slices[0].Name)
-	assert.Equal(t, svc.Name, slices[0].Labels[discovery.LabelServiceName])
-	assert.EqualValues(t, []discovery.EndpointPort{}, slices[0].Ports)
-	assert.EqualValues(t, []discovery.Endpoint{}, slices[0].Endpoints)
-	expectTrackedGeneration(t, r.endpointSliceTracker, &slices[0], 1)
-	expectMetrics(t, expectedMetrics{desiredSlices: 1, actualSlices: 1, desiredEndpoints: 0, addedPerSync: 0, removedPerSync: 0, numCreated: 0, numUpdated: 1, numDeleted: 0, slicesChangedPerSync: 1})
+			numActionsBefore := len(client.Actions())
+			r := newReconciler(client, []*corev1.Node{{ObjectMeta: metav1.ObjectMeta{Name: "node-1"}}}, defaultMaxEndpointsPerSlice)
+
+			// Run reconcile multiple times to make sure that subsequent calls don't do any additional work
+			for i := 0; i < 1; i++ {
+				reconcileHelper(t, r, &svc, []*corev1.Pod{}, existingSlices, time.Now())
+			}
+
+			var numUpdates int
+			if tc.wantUpdate {
+				numUpdates = 1
+			}
+			wantActions := numActionsBefore + numUpdates
+			assert.Len(t, client.Actions(), wantActions, "Expected %d additional clientset actions", numUpdates)
+
+			slices := fetchEndpointSlices(t, client, namespace)
+			assert.Len(t, slices, 1, "Expected 1 endpoint slices")
+
+			if !tc.wantUpdate {
+				assert.Regexp(t, "^"+svc.Name, slices[0].Name)
+				assert.Equal(t, svc.Name, slices[0].Labels[discovery.LabelServiceName])
+				assert.EqualValues(t, []discovery.EndpointPort{}, slices[0].Ports)
+				assert.EqualValues(t, []discovery.Endpoint{}, slices[0].Endpoints)
+				if tc.existing == nil {
+					expectTrackedGeneration(t, r.endpointSliceTracker, &slices[0], 1)
+				}
+			}
+			expectMetrics(t, tc.wantMetrics)
+		})
+	}
+}
+
+func TestSliceEqual(t *testing.T) {
+	testCases := []struct {
+		desc string
+		x    *discovery.EndpointSlice
+		y    *discovery.EndpointSlice
+		want bool
+	}{
+		{
+			desc: "Both nil",
+			want: true,
+		},
+		{
+			desc: "Y is nil",
+			x:    &discovery.EndpointSlice{},
+			want: false,
+		},
+		{
+			desc: "X is nil",
+			y:    &discovery.EndpointSlice{},
+			want: false,
+		},
+		{
+			desc: "Both are empty and non-nil",
+			x:    &discovery.EndpointSlice{},
+			y:    &discovery.EndpointSlice{},
+			want: true,
+		},
+		{
+			desc: "Both are empty and non-nil",
+			x:    &discovery.EndpointSlice{},
+			y:    &discovery.EndpointSlice{},
+			want: true,
+		},
+		{
+			desc: "Only ObjectMeta has diff",
+			x: &discovery.EndpointSlice{ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+			}},
+			y: &discovery.EndpointSlice{ObjectMeta: metav1.ObjectMeta{
+				Name: "bar",
```

Comment:
two slices with different names should be considered equal?

Reply:
In this case yes, since the placeholder slice will have a generated name that can change between resync periods